app/src/main/java/org/mifos/selfserviceapp/ui/activities/PassCodeActivity.java

```diff
@@ -88,15 +87,18 @@ public void savePassCode() {
         }
     }
 
-    @OnClick(R.id.btn_login)
-    public void loginUsingPassCode() {
+
+    @Override
+    public void passCodeEntered(String passcode) {
 
         if (!isInternetAvailable()) {
+            passCodeView.clearPasscodeField();
             return;
         }
 
         if (counter == 3) {
```

Comment:
@dilpreet96 I think you should write
if (counter == 3 && !preferencesHelper.getPasscode().equals(passwordEntered))

Right now if user has entered wrong password 3 time and user is entering correct password in 4th time then he is not able to login.